syntax = "proto3";

package ctp;

// 投资者持仓信息消息
message InvestorPositionMessage {
    // 合约代码
    string instrument_id = 1;
    // 经纪公司代码
    string broker_id = 2;
    // 投资者代码
    string investor_id = 3;
    // 持仓多空方向
    string posi_direction = 4;
    // 投机套保标志
    string hedge_flag = 5;
    // 持仓日期
    string position_date = 6;
    // 上日持仓
    int32 yd_position = 7;
    // 今日持仓
    int32 position = 8;
    // 多头冻结
    int32 long_frozen = 9;
    // 空头冻结
    int32 short_frozen = 10;
    // 开仓冻结金额
    double long_frozen_amount = 11;
    // 开仓冻结金额
    double short_frozen_amount = 12;
    // 开仓量
    int32 open_volume = 13;
    // 平仓量
    int32 close_volume = 14;
    // 开仓金额
    double open_amount = 15;
    // 平仓金额
    double close_amount = 16;
    // 持仓成本
    double position_cost = 17;
    // 上次占用的保证金
    double pre_margin = 18;
    // 占用的保证金
    double use_margin = 19;
    // 冻结的保证金
    double frozen_margin = 20;
    // 冻结的资金
    double frozen_cash = 21;
    // 冻结的手续费
    double frozen_commission = 22;
    // 资金差额
    double cash_in = 23;
    // 手续费
    double commission = 24;
    // 平仓盈亏
    double close_profit = 25;
    // 持仓盈亏
    double position_profit = 26;
    // 上次结算价
    double pre_settlement_price = 27;
    // 本次结算价
    double settlement_price = 28;
    // 交易日
    string trading_day = 29;
    // 结算编号
    int32 settlement_id = 30;
    // 开仓成本
    double open_cost = 31;
    // 交易所保证金
    double exchange_margin = 32;
    // 组合成交形成的持仓
    int32 comb_position = 33;
    // 组合多头冻结
    int32 comb_long_frozen = 34;
    // 组合空头冻结
    int32 comb_short_frozen = 35;
    // 逐日盯市平仓盈亏
    double close_profit_by_date = 36;
    // 逐笔对冲平仓盈亏
    double close_profit_by_trade = 37;
    // 今日持仓
    int32 today_position = 38;
    // 保证金率
    double margin_rate_by_money = 39;
    // 保证金率(按手数)
    double margin_rate_by_volume = 40;
    // 执行冻结
    int32 strike_frozen = 41;
    // 执行冻结金额
    double strike_frozen_amount = 42;
    // 放弃执行冻结
    int32 abandon_frozen = 43;
    // 交易所代码
    string exchange_id = 44;
    // 执行冻结的昨仓
    int32 yd_strike_frozen = 45;
    // 投资单元代码
    string invest_unit_id = 46;
    // 持仓成本差值
    double position_cost_offset = 47;
}

// 批量投资者持仓消息
message InvestorPositionBatchMessage {
    // 时间戳
    int64 timestamp = 1;
    // 消息类型
    string message_type = 2;
    // 持仓列表
    repeated InvestorPositionMessage positions = 3;
} 